lightning import parquet file fail for "runtime error: invalid memory address or nil pointer dereference"

## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

lightning import parquet file,
backend: tidb or local

### 2. What did you expect to see? (Required)
import successfully

### 3. What did you see instead (Required)
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0x4b6ef82]

goroutine 836 [running]:
github.com/pingcap/tidb/br/pkg/lightning/backend.(*ClosedEngine).Logger(...)
        /home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/br/br/pkg/lightning/backend/backend.go:392
github.com/pingcap/tidb/br/pkg/lightning/importer.(*TableImporter).importKV(0xc0012a0460, {0x6018d88, 0xc001081220}, 0x0, 0xc000ffe000)
        /home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/br/br/pkg/lightning/importer/table_import.go:1320 +0x42
github.com/pingcap/tidb/br/pkg/lightning/importer.(*TableImporter).importEngine(0xc000ffc1b0?, {0x6018d88?, 0xc001081220}, 0x0?, 0xc000ffe000, 0x0?)
        /home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/br/br/pkg/lightning/importer/table_import.go:918 +0x3a
github.com/pingcap/tidb/br/pkg/lightning/importer.(*TableImporter).importEngines.func3(0xc000a84fd0?, 0x0, 0xc00109ece0)
        /home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/br/br/pkg/lightning/importer/table_import.go:525 +0x2d3
created by github.com/pingcap/tidb/br/pkg/lightning/importer.(*TableImporter).importEngines in goroutine 719
        /home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/br/br/pkg/lightning/importer/table_import.go:516 +0x959
2023-12-12T15:11:08.656+0800    INFO    host/host.go:52 Execute command error   {"command": "/tidb-lightning \"-c\" \"/tmp/tidb-lightning.toml\"", "exit code": 2, "stdout": "Verbose debug logs will be written to /tmp/lightning.log.2023-12-12T07.11.07Z\n\n+---+----------------------------------------------+-------------+--------+\n| # | CHECK ITEM                                   | TYPE        | PASSED |\n+---+----------------------------------------------+-------------+--------+\n| 1 | Source data files size is proper             | performance | true   |\n+---+----------------------------------------------+-------------+--------+\n| 2 | the checkpoints are valid                    | critical    | true   |\n+---+----------------------------------------------+-------------+--------+\n| 3 | Cluster version check passed                 | critical    | true   |\n+---+----------------------------------------------+-------------+--------+\n| 4 | Lightning has the correct storage permission | critical    | true   |\n+---+----------------------------------------------+-------------+--------+\n\n", "stderr": "panic: runtime error: invalid memory address or nil pointer dereference\n[signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0x4b6ef82]\n\ngoroutine 836 [running]:\ngithub.com/pingcap/tidb/br/pkg/lightning/backend.(*ClosedEngine).Logger(...)\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/br/br/pkg/lightning/backend/backend.go:392\ngithub.com/pingcap/tidb/br/pkg/lightning/importer.(*TableImporter).importKV(0xc0012a0460, {0x6018d88, 0xc001081220}, 0x0, 0xc000ffe000)\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/br/br/pkg/lightning/importer/table_import.go:1320 +0x42\ngithub.com/pingcap/tidb/br/pkg/lightning/importer.(*TableImporter).importEngine(0xc000ffc1b0?, {0x6018d88?, 0xc001081220}, 0x0?, 0xc000ffe000, 0x0?)\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/br/br/pkg/lightning/importer/table_import.go:918 +0x3a\ngithub.com/pingcap/tidb/br/pkg/lightning/importer.(*TableImporter).importEngines.func3(0xc000a84fd0?, 0x0, 0xc00109ece0)\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/br/br/pkg/lightning/importer/table_import.go:525 +0x2d3\ncreated by github.com/pingcap/tidb/br/pkg/lightning/importer.(*TableImporter).importEngines in goroutine 719\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/br/br/pkg/lightning/importer/table_import.go:516 +0x959\n", "error": "command terminated with exit code 2"}
2023-12-12T15:11:08.720+0800    INFO    host/host.go:56 Execute command finished        {"command": "/tidb-lightning \"-c\" \"/tmp/tidb-lightning.toml\"", "execute duration": "2.133774334s"}
```
### 4. What is your TiDB version? (Required)

master

--- a/br/pkg/lightning/importer/table_import.go
+++ b/br/pkg/lightning/importer/table_import.go
@@ -771,8 +771,9 @@ ChunkLoop:
 
 		if chunk.FileMeta.Type == mydump.SourceTypeParquet {
 			// TODO: use the compressed size of the chunk to conduct memory control
-			_, err = getChunkCompressedSizeForParquet(ctx, chunk, rc.store)
-			return nil, errors.Trace(err)
+			if _, err = getChunkCompressedSizeForParquet(ctx, chunk, rc.store); err != nil {
+				return nil, errors.Trace(err)
+			}
 		}
 
 		restoreWorker := rc.regionWorkers.Apply()

